commit 094e3f7d091b35b171bd08cfe95ace5f9bdcc470
Author: tobhe <tobhe@openbsd.org>
Date:   Wed Jul 14 22:39:26 2021 +0000

    Export SA replay counters via pfkey and print with ipsecctl.
    This is useful for debugging replay window issues with 64 bit
    sequence numbers in IPsec.
    
    ok bluhm@

diff --git sbin/ipsecctl/pfkdump.c sbin/ipsecctl/pfkdump.c
index cc13c327c5c..f848a80a93a 100644
--- diff-019.c
+++ diff-019.c
@@ -1,4 +1,4 @@
-/*	$OpenBSD: pfkdump.c,v 1.53 2021/07/05 12:03:42 tobhe Exp $	*/
+/*	$OpenBSD: pfkdump.c,v 1.54 2021/07/14 22:39:26 tobhe Exp $	*/
 
 /*
  * Copyright (c) 2003 Markus Friedl.  All rights reserved.
@@ -57,6 +57,7 @@ static void	print_ident(struct sadb_ext *, struct sadb_msg *, int);
 static void	print_udpenc(struct sadb_ext *, struct sadb_msg *, int);
 static void	print_tag(struct sadb_ext *, struct sadb_msg *, int);
 static void	print_rdomain(struct sadb_ext *, struct sadb_msg *, int);
+static void	print_replay(struct sadb_ext *, struct sadb_msg *, int);
 static void	print_mtu(struct sadb_ext *, struct sadb_msg *, int);
 static void	print_tap(struct sadb_ext *, struct sadb_msg *, int);
 static void	print_satype(struct sadb_ext *, struct sadb_msg *, int);
@@ -109,6 +110,7 @@ struct idname ext_types[] = {
 	{ SADB_X_EXT_LIFETIME_LASTUSE,	"lifetime_lastuse",	print_life },
 	{ SADB_X_EXT_TAG,		"tag",			print_tag },
 	{ SADB_X_EXT_RDOMAIN,		"rdomain",		print_rdomain },
+	{ SADB_X_EXT_REPLAY,		"replay",		print_replay },
 	{ SADB_X_EXT_MTU,		"mtu",			print_mtu },
 	{ SADB_X_EXT_TAP,		"tap",			print_tap },
 	{ SADB_X_EXT_SATYPE2,		"satype2",		print_satype },
@@ -412,6 +414,14 @@ print_tag(struct sadb_ext *ext, struct sadb_msg *msg, int opts)
 	printf("%s", p);
 }
 
+static void
+print_replay(struct sadb_ext *ext, struct sadb_msg *msg, int opts)
+{
+	struct sadb_x_replay *sreplay = (struct sadb_x_replay *)ext;
+
+	printf("rpl %llu", sreplay->sadb_x_replay_count);
+}
+
 static void
 print_mtu(struct sadb_ext *ext, struct sadb_msg *msg, int opts)
 {
